recipes-bsp/bootfiles/rpi-config_git.bb

```diff
@@ -115,8 +115,7 @@ do_deploy() {
 
     # Video camera support
     if [ "${VIDEO_CAMERA}" = "1" ]; then
-        echo "# Enable video camera" >>${DEPLOYDIR}/bcm2835-bootfiles/config.txt
-        echo "start_x=1" >>${DEPLOYDIR}/bcm2835-bootfiles/config.txt
+        sed -i '/#start_x=/ c\start_x=1' ${DEPLOYDIR}/bcm2835-bootfiles/config.txt
```

Comment:
Do we have any explanation for this? How is the config in Raspbian?

Reply:
it needs gpu memory >= 128M to be reserved, I am fine with enabling this but perhaps it should be a knob like ENABLE_UART and perhaps kept off by default, and clearly documented how to enable camera with meta-raspberrypi

Reply:
@agherzan, I have no explanations on this. Doing some test, I noticed that with my raspberry pi 4b and with my SD card, if I put the "start_x=1" after the ~643 line, I have no access to the camera when the system boots up. 

The config.txt of the raspbian image I have, has only ~70 lines. This makes me think that there could be a restriction in the 
config file size.

@kraj There is already a knob for this: VIDEO_CAMERA = "1" and is documented in docs/extra-build-config.md. What I did is instead of adding "start_x=1" at the end of the file (line ~1187), replace the already commented "#start_x=0" (line 469) with "start_x=1". I forgot to mention that I'm using "gpu_mem=512".

Would be nice if anyone can replicate what I'm seeing, it's a bit strange for me.

Reply:
In that case, it sounds like something else is going on here. Size could be a suspect.

Reply:
I can confirm that this patch fixes the camera problem for me as well on RPi 4B (4GB)

Reply:
I'm a bit uneasy at the fact that we don't really understand what is going on here. I get that this is fixing an issue but understanding if this is a configuration size limitation, is  important. It could actually reveal an issue because our config has tons of comments. If that gets confirmed, we should clean that up a bit.

Reply:
I don't think this is a configuration size limitation. I added some more HDMI configurations at the end, and they are picked up without any problems. It's just something special about start_x (I don't know what, yet).

If it makes any difference: all my testing is on u-boot.

Reply:
`start_x` is actually special since its an alias  to `start_file=start_x.elf fixup_file=fixup_x.dat`  so I wonder if any 
of these options are changed in between this option being set/rest for more details see [1]

[1] https://www.raspberrypi.org/documentation/configuration/config-txt/boot.md

Reply:
@gvkalra Have you managed to find more info on this? If not, Let's get this in (so we fix the issue at hand) but also add a `TODO` comment to flag this issue in the recipe.

Reply:
Unfortunately, I don't have more information on this ¯\_(ツ)_/¯

Reply:
Ok. Let's throw a comment in the recipe explaining the conundrum and proceed with the merge.

Reply:
Added the TODO with a comment telling that the cause can be the file size, as I still belive it after doing some more tests.

I coudn't identify that the order of the variables in the file is creating this issue.

And working with the Raspbian config file, if I add an array of '#' caracteres of size 150x150 before the "start_x=1", the camera is not enabled. The resulting file looks like the following ("start_x=1" is at the end):

```
# For more options and information see
# http://rpf.io/configtxt
# Some settings may impact device functionality. See link above for details

# uncomment if you get no picture on HDMI for a default "safe" mode
#hdmi_safe=1

# uncomment this if your display has a black border of unused pixels visible
# and your display can output without overscan
#disable_overscan=1

# uncomment the following to adjust overscan. Use positive numbers if console
# goes off screen, and negative if there is too much border
#overscan_left=16
#overscan_right=16
#overscan_top=16
#overscan_bottom=16

# uncomment to force a console size. By default it will be display's size minus
# overscan.
#framebuffer_width=1280
#framebuffer_height=720

# uncomment if hdmi display is not detected and composite is being output
#hdmi_force_hotplug=1

# uncomment to force a specific HDMI mode (this will force VGA)
#hdmi_group=1
#hdmi_mode=1

# uncomment to force a HDMI mode rather than DVI. This can make audio work in
# DMT (computer monitor) modes
#hdmi_drive=2

# uncomment to increase signal to HDMI, if you have interference, blanking, or
# no display
#config_hdmi_boost=4

# uncomment for composite PAL
#sdtv_mode=2

#uncomment to overclock the arm. 700 MHz is the default.
#arm_freq=800

# Uncomment some or all of these to enable the optional hardware interfaces
#dtparam=i2c_arm=on
#dtparam=i2s=on
#dtparam=spi=on

# Uncomment this to enable infrared communication.
#dtoverlay=gpio-ir,gpio_pin=17
#dtoverlay=gpio-ir-tx,gpio_pin=18

# Additional overlays and parameters are documented /boot/overlays/README

# Enable audio (loads snd_bcm2835)
dtparam=audio=on

[pi4]
# Enable DRM VC4 V3D driver on top of the dispmanx display stack
dtoverlay=vc4-fkms-v3d
max_framebuffers=2

enable_uart=1
gpu_mem=512

[all]
#dtoverlay=vc4-fkms-v3d
###############################             ...             ########################################
...
...  150x150 of '#'
...
###############################             ...             ########################################
start_x=1

```

This array of '#' has 22.2 kB in size which is even less than the config file meta-raspberrypi is using.

I'll open a thread in the raspberrypi.org forum asking this, and I'll open a new pull request to remove the coment if I get an answer.

Reply:
Looks good.